packages/editor/src/bindings/entity-url.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { store as coreDataStore } from '@wordpress/core-data';
+
+export default {
+	name: 'core/entity-url',
+	label: __( 'Entity URL' ),
+	getValues( { select, clientId } ) {
+		const { getBlockAttributes } = select( 'core/block-editor' );
+
+		// Get the nav link's id attribute
+		const blockAttributes = getBlockAttributes( clientId );
+		const linkedPostId = blockAttributes?.id;
+
+		if ( ! linkedPostId ) {
+			return {};
+		}
+
+		const { getEntityRecord } = select( coreDataStore );
+
+		// Get the post type and kind from block attributes
+		const { type, kind } = blockAttributes || {};
+
+		let url = '';
+
+		// Handle post types
+		if ( kind === 'post-type' ) {
+			const post = getEntityRecord(
+				'postType',
+				type || 'post',
+				linkedPostId
+			);
+			url = post?.link || '';
+		}
+		// Handle taxonomies
+		else if ( kind === 'taxonomy' ) {
+			// Convert 'tag' back to 'post_tag' for API calls
+			// See update-attributes.js line 166 for the reverse conversion
+			const taxonomySlug = type === 'tag' ? 'post_tag' : type;
+			const term = getEntityRecord(
+				'taxonomy',
+				taxonomySlug,
+				linkedPostId
+			);
+			url = term?.link || '';
+		}
+
```

Comment:
We probably need to magically handle `post-type-archive` links as well if that's something we even want to do.

For example, if you link to the `Books` Post archive and then you rename the CPT to `Paper` should you expect the link to automatically update or would that be taking things too far?